ui/components/app/assets/nfts/nfts-tab/nfts-tab.js

```diff
@@ -93,6 +94,12 @@ export default function NftsTab() {
     currentLocale,
   ]);
 
+  useEffect(() => {
+    if (!nftsLoading && !nftsStillFetchingIndication) {
+      endTrace({ name: TraceName.AccountOverviewNftsTab });
+    }
+  }, [nftsLoading, nftsStillFetchingIndication]);
+
```

Comment:
From https://react.dev/learn/you-might-not-need-an-effect#recap
> If you can calculate something during render, you don’t need an Effect.

`useEffect` to run code "when this variable changes" is done **A LOT**, but it's actually an anti-pattern

Reply:
```suggestion
  if (!nftsLoading && !nftsStillFetchingIndication) {
    endTrace({ name: TraceName.AccountOverviewNftsTab });
  }

```

Reply:
Applied, but with some questions: Wouldn't this be a textbook case where `useEffect` is warranted since we're interfacing with an external/non-local system and triggering a side effect that depends on reactive variables? 

I'm not sure if introducing impure logic to the top-level scope of a react component is in keeping with the article's recommendations. This usage also doesn't seem to fall under any of the example cases discussed in the article (no computations, no altering state in this or any relative component, no event handlers, etc.).

That said, removing the `useEffect` hooks doesn't seem to trigger extra re-renders or cause malfunctions, so this might be more a question of principle rather than practicality. Still would be nice to understand what the best practice is here.

Reply:
I reviewed the react docs, and it seems to indicate that `trace`/`endTrace` calls not triggered by events should always be placed inside `useEffect` hooks:

1. `trace`, `endTrace` are side effects that synchronize with an external system (i.e. sentry account). They're also implemented as POST requests, which are definitely Effects.

> Some components need to synchronize with external systems. For example, you might want to ... *send an analytics log when a component appears on the screen.* Effects let you run some code after rendering so that you can synchronize your component with some system outside of React.
> — https://react.dev/learn/synchronizing-with-effects#sending-analytics

2. These `trace`, `endTrace` calls are solely motivated by the component being displayed, and are conditionally dependent on reactive values in the component's rendering logic. 

> When you’re not sure whether some code should be in an Effect or in an event handler, ask yourself why this code needs to run. **Use Effects only for code that should run because the component was displayed to the user.** In this example, the notification should appear because the user pressed the button, not because the page was displayed!
> — https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers

> The analytics POST request should remain in an Effect. This is because *the reason to send the analytics event is that the form was displayed*.
> — https://react.dev/learn/you-might-not-need-an-effect#sending-a-post-request

3. [`useEffect` works by reacting to reactive values](https://react.dev/learn/lifecycle-of-reactive-effects#effects-react-to-reactive-values), and [`useEffect` dependencies are reactive values](https://react.dev/learn/lifecycle-of-reactive-effects#react-verifies-that-you-specified-every-reactive-value-as-a-dependency), meaning an effect triggering a re-render in response to a reactive value being updated is expected behavior, not a performance penalty. We shouldn't be removing `useEffect` just because they cause re-renders, without examining whether they're doing so unnecessarily, or as part of the hook's core functionality.

4. Re-renders aren't the only performance consideration here. Removing `useEffect` makes its inner code block re-run on every render instead of only when the dependencies are updated. Avoiding these wasteful re-runs is the purpose of populating the dependency array in `useEffect` hooks.